### PR TITLE
Add the py-celery package

### PR DESCRIPTION
**Pull request by koning**
This package is a requirement for py-merlinwf

### CODE REVIEW

**Comment by koning** [comment]
This package depends on other PRs including:
#13848 
#13849 
#13855 
#13857

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    # 'librabbitmq',
+    # 'pyro',
+    # 'slmq',
+    # 'tblib',
+    # 'consul',
+    # 'dynamodb',
+    # 'mongodb',
+    # 'cosmosdbsql',
+    # 's3',
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```python
depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    depends_on('py-setuptools', type=('build', 'run'))
+
+    depends_on('librabbitmq', when='+rabbitmq', type='run')
+    depends_on('py-redis', when='+redis', type='run')
+
+    depends_on('py-pytz', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    depends_on('py-pytz@dev:', type=('build', 'run'))
```
Not sure if these means the develop version or what. As long as you don't see setuptools automatically downloading a newer version, we should be fine.

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class PyCelery(PythonPackage):
+    """Celery - Distributed Task Queue."""
+
+    homepage = "https://pypi.org/project/celery/"
+    url      = "https://pypi.io/packages/source/c/celery/celery-4.2.1.tar.gz"
+
+    version('4.3.0', sha256='4c4532aa683f170f40bd76f928b70bc06ff171a959e06e71bf35f2f9d6031ef9')
+    version('4.2.1', sha256='77dab4677e24dc654d42dfbdfed65fa760455b6bb563a0877ecc35f4cfcfc678')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]: This older version has different dependency requirements. I would drop it if you don't need it.

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    # 'cosmosdbsql',
+    # 's3',
+
+    depends_on('py-setuptools', type=('build', 'run'))
+
+    depends_on('librabbitmq', when='+rabbitmq', type='run')
+    depends_on('py-redis', when='+redis', type='run')
+
+    depends_on('py-pytz', type='run')
+    depends_on('py-billiard@3.6.0.0:', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    depends_on('py-billiard@3.6.0:3.999', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    depends_on('py-kombu@4.3.0:', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    depends_on('py-kombu@4.4.0:4.999', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    depends_on('py-vine@1.3.0:', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    depends_on('py-vine@1.3.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+class PyCelery(PythonPackage):
+    """Celery - Distributed Task Queue."""
+
+    homepage = "https://pypi.org/project/celery/"
+    url      = "https://pypi.io/packages/source/c/celery/celery-4.2.1.tar.gz"
+
+    version('4.3.0', sha256='4c4532aa683f170f40bd76f928b70bc06ff171a959e06e71bf35f2f9d6031ef9')
+    version('4.2.1', sha256='77dab4677e24dc654d42dfbdfed65fa760455b6bb563a0877ecc35f4cfcfc678')
+
+    variant('rabbitmq', default=False, description="Use rabbitMQ transport")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    variant('librabbitmq', default=False, description="Use rabbitMQ transport")
```

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    # 'tblib',
+    # 'consul',
+    # 'dynamodb',
+    # 'mongodb',
+    # 'cosmosdbsql',
+    # 's3',
+
+    depends_on('py-setuptools', type=('build', 'run'))
+
+    depends_on('librabbitmq', when='+rabbitmq', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    depends_on('py-librabbitmq@1.5.0:', when='+librabbitmq', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-celery/package.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    depends_on('py-redis', when='+redis', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-celery/package.py]:
```suggestion
    depends_on('py-redis@3.2.0:', when='+redis', type=('build', 'run'))
```

**Pull request merged**